packages/functional/src/__tests__/pipe-transaction-test.ts

```diff
@@ -0,0 +1,325 @@
+            expect(tx2.version).toBe(0);
+            expect(tx2.lifetimeConstraint).toBe(blockhashLifetime);
+            expect(tx2.feePayer).toBe(feePayer);
+        });
+        describe('when more than one fee payer is set in the same pipe', () => {
+            it('uses the last fee payer', () => {
+                expect.assertions(6);
+                const tx1 = pipeTransaction(
+                    // Commenting for legibility
+                    createTransaction({ version: 'legacy' }),
+                    tx => setTransactionFeePayer(feePayer, tx),
+                    tx => setTransactionLifetimeUsingBlockhash(blockhashLifetime, tx),
+                    tx => setTransactionFeePayer(secondFeePayer, tx)
+                );
+                const tx2 = pipeTransaction(
+                    // Commenting for legibility
+                    createTransaction({ version: 0 }),
+                    tx => setTransactionFeePayer(feePayer, tx),
+                    tx => setTransactionLifetimeUsingBlockhash(blockhashLifetime, tx),
+                    tx => setTransactionFeePayer(secondFeePayer, tx)
+                );
+                expect(tx1.version).toBe('legacy');
+                expect(tx1.lifetimeConstraint).toBe(blockhashLifetime);
+                expect(tx1.feePayer).toBe(secondFeePayer);
+                expect(tx2.version).toBe(0);
+                expect(tx2.lifetimeConstraint).toBe(blockhashLifetime);
+                expect(tx2.feePayer).toBe(secondFeePayer);
+            });
+        });
+        describe('when more than one blockhash is set in the same pipe', () => {
+            it('uses the last blockhash', () => {
+                expect.assertions(6);
+                const tx1 = pipeTransaction(
+                    // Commenting for legibility
+                    createTransaction({ version: 'legacy' }),
+                    tx => setTransactionLifetimeUsingBlockhash(blockhashLifetime, tx),
+                    tx => setTransactionFeePayer(feePayer, tx),
+                    tx => setTransactionLifetimeUsingBlockhash(secondBlockhashLifetime, tx)
+                );
+                const tx2 = pipeTransaction(
+                    // Commenting for legibility
+                    createTransaction({ version: 0 }),
+                    tx => setTransactionLifetimeUsingBlockhash(blockhashLifetime, tx),
+                    tx => setTransactionFeePayer(feePayer, tx),
+                    tx => setTransactionLifetimeUsingBlockhash(secondBlockhashLifetime, tx)
+                );
+                expect(tx1.version).toBe('legacy');
+                expect(tx1.lifetimeConstraint).toBe(secondBlockhashLifetime);
+                expect(tx1.feePayer).toBe(feePayer);
+                expect(tx2.version).toBe(0);
+                expect(tx2.lifetimeConstraint).toBe(secondBlockhashLifetime);
+                expect(tx2.feePayer).toBe(feePayer);
+            });
+        });
+    });
+});
```

Comment:
Are these tests the same as those in `pipe-test.ts`?

It might make sense to do non-transaction-specific tests for `pipe-test` instead IMO, which would also make it easier to follow for someone using `pipe` for something else.

Reply:
Yes, and yes you're correct I was just lazy lol. This isn't ready for merge yet